### PR TITLE
new: Upgrade Beemo and rewrite in TS.

### PR DESCRIPTION
**Pull request by milesj**
to: @hayes @williaster 

Beemo has been updated upstream to support compiled `lib/` paths, so now we have the opportunity to write our source in TS. During this refactor, I discovered a handful of bugs.

### CODE REVIEW

File: packages/config-eslint/src/resolvers/graphql.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { GQL_EXT_PATTERN } from '@airbnb/nimbus-common';
+
+export const interfaceVersion = 2;
+
+export function resolve(source: string): { found: boolean; path?: unknown } {
```
Comment (williaster) [inline packages/config-eslint/src/resolvers/graphql.ts]: why `path?: unknown` vs `null`?

File: packages/config-webpack/src/helpers.ts [hunk 2]
```diff
@@ -0,0 +1,102 @@
+// import fs from 'fs';
+// import path from 'path';
```
Comment (williaster) [inline packages/config-webpack/src/helpers.ts]: remove?

File: packages/nimbus-common/src/core.ts [hunk 3]
```diff
@@ -0,0 +1,108 @@
+import fs from 'fs';
+import path from 'path';
+import execa from 'execa';
+import glob from 'fast-glob';
+import Beemo, { BeemoConfig } from '@beemo/core';
+import { PackageConfig } from '@boost/core';
+
+export interface NimbusEnvSetting {
```
Comment (williaster) [inline packages/nimbus-common/src/core.ts]: these types are gonna be SO useful 🥇

File: packages/config-eslint/src/resolvers/graphql.ts (hunk 1 above)
Comment (milesj) [reply packages/config-eslint/src/resolvers/graphql.ts]: Just because I don't know what eslint expects it to be.

**Pull request merged**